### PR TITLE
feat: retriever cookbook and utils change

### PR DESCRIPTION
**Pull request by Wendong-Fan**
## Description

This this a sub PR of https://github.com/camel-ai/camel/pull/410 only containing retriever cookbook and utils change based on auto retriever, with a small typo fix

## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/camel-ai/camel/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Implemented Tasks

- [ ] Subtask 1
- [ ] Subtask 2
- [ ] Subtask 3

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [ ] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: camel/utils/commons.py [hunk 1]
```diff
@@ -29,6 +29,17 @@
...
 
+# Set lazy import
+def get_lazy_imported_functions_module():
+    from camel.functions import MATH_FUNCS, SEARCH_FUNCS, WEATHER_FUNCS
+    return [*MATH_FUNCS, *SEARCH_FUNCS, *WEATHER_FUNCS]
+
+
+def get_lazy_imported_types_module():
+    from camel.types import ModelType
+    return ModelType.GPT_4_TURBO
```
Comment (Appointat) [inline camel/utils/commons.py:40]:
```suggestion
# Set lazy import
def get_lazy_imported_types_module():
    from camel.types import ModelType
    return ModelType.GPT_4_TURBO
```

File: camel/utils/commons.py [hunk 2]
```diff
@@ -226,3 +237,129 @@ def to_pascal(snake: str) -> str:
...
+        model_type = get_lazy_imported_types_module()
+
+    from colorama import Fore
+
+    from camel.agents.chat_agent import FunctionCallingRecord
+    from camel.configs import ChatGPTConfig, FunctionCallingConfig
+    from camel.societies import RolePlaying
+
+    task_prompt = task_prompt
+    user_model_config = ChatGPTConfig(temperature=0.0)
```
Comment (Appointat) [inline camel/utils/commons.py:295]:
```suggestion
    task_prompt = task_prompt

    user_model_config = ChatGPTConfig(temperature=0.0)
```

File: camel/utils/commons.py [hunk 3]
```diff
@@ -226,3 +237,129 @@ def to_pascal(snake: str) -> str:
...
+            break
+
+        # Print output from the user
+        print_text_animated(Fore.BLUE +
+                            f"AI User:\n\n{user_response.msg.content}\n")
+
+        # Print output from the assistant, including any function
+        # execution information
+        print_text_animated(Fore.GREEN + "AI Assistant:")
+        called_functions: List[
```
Comment (Appointat) [inline camel/utils/commons.py:355]:
```suggestion
        print_text_animated(Fore.GREEN + "AI Assistant:")

        called_functions: List[
```

**Review by Appointat — APPROVED** [review]
I've reviewed the code and left some simple modification suggestions (some regarding formatting). Well, I'm not sure if I missed anything in the dev meeting. Could you please explain to me why you're moving the `role_playing...` function into `commons.py`? Thanks

**Comment by Wendong-Fan** [comment]
> I've reviewed the code and left some simple modification suggestions (some regarding formatting). Well, I'm not sure if I missed anything in the dev meeting. Could you please explain to me why you're moving the `role_playing...` function into `commons.py`? Thanks

Hey @Appointat , thanks for the review, in the cookbook I used `role_playing...` function as example, I think this function will be regularly used as a common example function to show how to implement tasks with multi-agents, so I moved this function into utils

File: camel/utils/commons.py [hunk 4]
```diff
@@ -226,3 +238,131 @@ def to_pascal(snake: str) -> str:
...
+            chat session. Defaults to 10.
+        assistant_role_name (str): The role name assigned to the AI Assistant.
+            Defaults to 'Searcher'.
+        user_role_name (str): The role name assigned to the User. Defaults to
+            'Professor'.
+
+    Returns:
+        None: This function does not return any value but prints out the
+            session's dialogues and outputs.
+    """
```
Comment (lightaime) [inline camel/utils/commons.py:281]: Why the `role_playing_with_function`  example is moved to `utils`?

Comment (lightaime) [reply camel/utils/commons.py:281]: Let's not put hard coded example into `utils`.

Comment (Wendong-Fan) [reply camel/utils/commons.py:281]: > Why the `role_playing_with_function` example is moved to `utils`?

Hey @lightaime , in the RAG cookbook I used `role_playing_with_function` as example,


````
from camel.utils import role_playing_with_function
from camel.functions import MATH_FUNCS  # import another function from camel

role_playing_with_function(
    task_prompt=(
        "What is the main termination reasons for AI Society dataset, how many number of messages did camel decided to limit, what's the value plus 100?"
    ),
    function_list=[*RETRIEVER_FUNCS, *MATH_FUNCS],
)
````

 I think this function will be regularly used as a common example function to show how to implement tasks by using different functions with multi-agents, so I moved this function into utils, even this function has set default values, it still allows the user to input their own parameters including `task_prompt`, `function_list`,`model_type`, `chat_turn_limit`, `assistant_role_name`,`user_role_name` to adjust the behavior of this function with flexibility.

**Pull request merged**